src/main/java/com/example/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.order.CreateOrderRequestDto;
+import com.example.onlinebookstore.dto.order.OrderDto;
+import com.example.onlinebookstore.dto.orderitem.OrderItemDto;
+import com.example.onlinebookstore.mapper.OrderItemMapper;
+import com.example.onlinebookstore.mapper.OrderMapper;
+import com.example.onlinebookstore.model.CartItem;
+import com.example.onlinebookstore.model.Order;
+import com.example.onlinebookstore.model.OrderItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.order.OrderRepository;
+import com.example.onlinebookstore.repository.orderitem.OrderItemRepository;
+import com.example.onlinebookstore.repository.shoppingcart.ShoppingCartRepository;
+import com.example.onlinebookstore.service.OrderService;
+import com.example.onlinebookstore.service.ShoppingCartService;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartService shoppingCartService;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Transactional
+    @Override
+    public OrderDto createOrder(User user, CreateOrderRequestDto requestDto) {
+        Order order = new Order();
+        ShoppingCart shoppingCart = shoppingCartRepository.findById(user.getId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find the shopping cart by id " + user.getId()));
+        List<OrderItem> orderItems = shoppingCart.getCartItems()
+                .stream()
+                .map(c -> convertToOrderItem(c, order))
+                .toList();
+        order.setOrderItems(orderItems);
+        order.setUser(user);
+        order.setOrderDate(LocalDateTime.now());
+        order.setStatus(Order.Status.PENDING);
+
+        BigDecimal total = orderItems.stream()
+                .map(o -> o.getBook().getPrice().multiply(BigDecimal.valueOf(o.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+        order.setTotal(total);
+        order.setShippingAddress(requestDto.getShippingAddress());
+        Order saved = orderRepository.save(order);
+        shoppingCartService.cleanShoppingCart(shoppingCart.getId());
+        return orderMapper.toDto(saved);
+    }
+
+    @Override
+    public List<OrderDto> getAll(Long userId) {
```

Comment:
pagination was not implemented
![image](https://github.com/waspp01/online-book-store-app/assets/73041091/91486adf-1331-4345-aeb5-49453e89b5bb)